### ISSUE TITLE
[question] Build subfolder is useless 

### ISSUE THREAD

**Issue by intelligide** [conan-io/conan-center-index#3126]
Some packages using cmake (or other build system) use a build subfolder. I think it is totally useless and may cause some path length problems on Windows (Paths are limited to 256 characters by default).

1. Sources are located in source_subfolder. So build files and sources can't conflict (Except if the source folder contains already a `source_subfolder`).
2. Using build_subfolder adds 15 extras characters to the path. The build folder already contains around 90 characters and CMake creates a huge and long file structure. It may cause some path length problems on Windows.

**Comment by uilianries** [comment]
Not really, build_subfolder isolates all artifacts generated when building. We decided to use it since Bincrafters, because we can check that folder, not mixing with the sources, that's important when you want to validate the copied binaries.

About Windows max path, build_folder is not the problem and it won't fix the situation, we solve for some recipes, but large names will have the same problem. There are other alternatives, like https://docs.microsoft.com/es-es/windows/win32/fileio/naming-a-file#maximum-path-length-limitation. Or it should be fixed in CCI internally, reducing the CI build path for instance.

**Comment by SpaceIm** [comment]
Long path is not available on all Windows machines (due to internal company rules for example). I think that reducing the chance of hitting path length limitation is more important than isolating artifacts which is just a nice to have for build checks.

**Comment by Croydon** [comment]
Is the path length a theoretical problem or did someone of you actual encounter a situation where
  * using the `short_paths=True` attribute did not help
  * and 15 characters less would have make a difference?

**Comment by SpaceIm** [comment]
Yes and `short_paths=True` is terrible on Windows, it's a mess to navigate in short paths and check build/source/package folder. And I can't even globally disable it when long path is enabled at Windows level, because of qt recipe...

**Comment by Croydon** [comment]
How many characters was it too much and was it a public recipe?

We could change the actual directory names from `source_subfolder` -> `source`  and  `build_subfolder` -> `build` as an intermediate step. This would reduce it by 10 characters

Otherwise I agree with @uilianries and would like to keep them. I also doubt that 5 (or even 15) characters would make a difference in many cases

**Comment by uilianries** [comment]
Renaming `build_subfolder` sounds plausible.

**Comment by Nekto89** [comment]
I currently have problem with HDF5. My Jenkins workspace adds around 50 symbols ...\.conan\data\hdf5\1.12.2\user\channel\build\85ef4fd94289a25d21b9b066eef368eb3151b158\build_subfolder\source_subfolder\utils\mirror_vfd\mirror_server_stop.dir\Release\mirror_s.37E5FFF9.tlog\link-cvtres.read.1.tlog
also custom user/channel might lead to additional length restrictions

Is it better to reduce length of build_subfolder/source_subfolder or just enable short_paths?
